### PR TITLE
Inflection

### PR DESCRIPTION
**Pull request by ralucarietkerk**
This PR adds inflect option for generating endpoints.

### CODE REVIEW

File: src/Model/Endpoint.php [hunk 1]
```diff
@@ -453,6 +464,22 @@ public function resourceClass($name = null)
     }
 
     /**
+     * Returns the inflect method or sets a new one
+     *
+     * @param null $method
+     *
+     * @return null|string
+     */
+    public function inflect($method = null)
```
Comment (ADmad) [inline src/Model/Endpoint.php]: `inflectionMethod` would be a better name. Currently the method name sounds as if it inflects string.

File: src/Model/Endpoint.php [hunk 2]
```diff
@@ -453,6 +464,22 @@ public function resourceClass($name = null)
     }
 
     /**
+     * Returns the inflect method or sets a new one
+     *
+     * @param null $method
```
Comment (ADmad) [inline src/Model/Endpoint.php]: https://travis-ci.org/UseMuffin/Webservice/jobs/295819481#L568

**Comment by ralucarietkerk** [comment]
The requested changes have been made.

File: src/Model/Endpoint.php [hunk 3]
```diff
@@ -109,6 +109,13 @@ class Endpoint implements RepositoryInterface, EventListenerInterface, EventDisp
     protected $_alias;
 
     /**
+     * The inflect method to use for endpoint routes
+     *
+     * @var string
+     */
+    protected $_inflection = 'underscore';
```
Comment (ADmad) [inline src/Model/Endpoint.php]: `$_inflectionMethod` please.

File: src/Model/Endpoint.php [hunk 4]
```diff
@@ -153,6 +160,9 @@ public function __construct(array $config = [])
         if (!empty($config['resourceClass'])) {
             $this->resourceClass($config['resourceClass']);
         }
+        if (!empty($config['inflection'])) {
```
Comment (ADmad) [inline src/Model/Endpoint.php]: Keep the option name `inflect` so that it's consistent with config name used in route builder.

Comment (ralucarietkerk) [reply src/Model/Endpoint.php]: thanks for the clarification!

**Comment by ralucarietkerk** [comment]
Wow, my first attempt at open-source contribution. Nice! thanks a lot!!

**Comment by ADmad** [comment]
Thank you for contributing. New release has been made which includes this change.

**Pull request merged**